client/components/theme-tier/theme-tier-badge/theme-tier-partner-badge.js

```diff
@@ -1,172 +1,99 @@
-import {
-	PLAN_BUSINESS,
-	getPlan,
-	PLAN_ECOMMERCE_TRIAL_MONTHLY,
-	PLAN_ECOMMERCE,
-} from '@automattic/calypso-products';
 import { PremiumBadge } from '@automattic/components';
-import { localizeUrl } from '@automattic/i18n-utils';
-import { createInterpolateElement } from '@wordpress/element';
+import clsx from 'clsx';
 import { useTranslate } from 'i18n-calypso';
+import { useMemo } from 'react';
 import { useSelector } from 'calypso/state';
-import { getSitePlanSlug } from 'calypso/state/sites/selectors';
 import { useIsThemeAllowedOnSite } from 'calypso/state/themes/hooks/use-is-theme-allowed-on-site';
 import {
 	isMarketplaceThemeSubscribed,
 	getMarketplaceThemeSubscriptionPrices,
 } from 'calypso/state/themes/selectors';
 import { getSelectedSiteId } from 'calypso/state/ui/selectors';
-import ThemeTierBadgeCheckoutLink from './theme-tier-badge-checkout-link';
 import { useThemeTierBadgeContext } from './theme-tier-badge-context';
-import ThemeTierTooltipTracker from './theme-tier-tooltip-tracker';
+import ThemeTierPlanUpgradeBadge from './theme-tier-upgrade-badge';
+
+const commonBadgeProps = {
+	focusOnShow: false,
+	isClickable: false,
+	shouldHideTooltip: true,
+	shouldHideIcon: false,
+};
 
-export default function ThemeTierPartnerBadge() {
+export default function ThemeTierPartnerBadge( {
+	showPartnerPrice,
+	hideBackgroundOnUpgrade,
+	isLongLabel,
+	hidePartnerBadge,
+} ) {
 	const translate = useTranslate();
 	const siteId = useSelector( getSelectedSiteId );
-	const planSlug = useSelector( ( state ) => getSitePlanSlug( state, siteId ) ?? '' );
+	const { themeId } = useThemeTierBadgeContext();
 
-	const isEcommerceTrialMonthly = planSlug === PLAN_ECOMMERCE_TRIAL_MONTHLY;
-	const planTooltipName = isEcommerceTrialMonthly ? 'ecommerce' : 'business';
-
-	const { showUpgradeBadge, themeId } = useThemeTierBadgeContext();
-	const isPartnerThemePurchased = useSelector( ( state ) =>
-		siteId ? isMarketplaceThemeSubscribed( state, themeId, siteId ) : false
+	const isPartnerThemePurchased = useSelector(
+		( state ) => ( siteId ? isMarketplaceThemeSubscribed( state, themeId, siteId ) : false ),
+		( prev, next ) => prev === next
```

Comment:
Nitpick (non-blocking): This is unnecessary, since it's the default behavior
```suggestion
		( state ) => ( siteId ? isMarketplaceThemeSubscribed( state, themeId, siteId ) : false )
```